### PR TITLE
[Remove Stream/Optional: Phase 4] LPS-182670  dynamic-data-mapping-service (Part 1)

### PR DESCRIPTION
**Pull request by JanisZhang**
https://issues.liferay.com/browse/LPS-182670

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:forms

**Comment by Kvale1733** [comment]
@JanisZhang No regressions.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/query/contributor/DDMFormInstanceRecordModelPreFilterContributor.java [hunk 1]
```diff
@@ -83,33 +80,22 @@ public void contribute(
...
-									structureId, notEmptyField, locale)),
-							BooleanClauseOccur.MUST));
-
-					booleanFilter.add(
-						notEmptyFieldBooleanFilter,
-						BooleanClauseOccur.MUST_NOT);
+			for (String notEmptyField : notEmptyFields) {
+				BooleanFilter notEmptyFieldBooleanFilter = new BooleanFilter();
+
+				for (String languageId : languageIds) {
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/query/contributor/DDMFormInstanceRecordModelPreFilterContributor.java:84]: @JanisZhang The `LocaleUtil.fromLanguageId` method looks a bit heavy. In the original logic, the `fromLanguageId` is called `languageIds.length` times; while in your logic, it's called `languageIds.length * notEmptyFields.length` times. In most occasions it's heavier than before. Please keep it a 2-step operation, thanks!

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@JanisZhang Please resend, thanks!

**Pull request closed without merging**